Proiect3D/11_01_Shader.frag

```diff
@@ -48,6 +48,78 @@ void main(void)
 	out_Color = vec4(result, 1.0f);
     }
 
+    if (codCol==2) // rosu
+    {
+  	// Ambient
+    float ambientStrength = 0.2f;
+    vec3 ambient = ambientStrength * lightColor;
+  	
+    // Diffuse 
+    vec3 normala = normalize(Normal);
+    vec3 lightDir = normalize(inLightPos - FragPos);
+    //vec3 lightDir = normalize(dir); // cazul unei surse directionale
+    float diff = max(dot(normala, lightDir), 0.0);
+    vec3 diffuse = diff * lightColor;
+    
+    // Specular
+    float specularStrength = 0.5f;
+    vec3 viewDir = normalize(inViewPos - FragPos);//vector catre observator normalizat (V)
+    vec3 reflectDir = reflect(-lightDir, normala); // reflexia razei de lumina (R)
+    float spec = pow(max(dot(viewDir, reflectDir), 0.0), 1);
+    vec3 specular = specularStrength * spec * lightColor;  
+    vec3 emission=vec3(0.0, 0.0, 0.0);
+    vec3 result = emission+(ambient + diffuse + specular) * vec3(1.0f, 0.0f, 0.0f);
+	out_Color = vec4(result, 1.0f);
+    }
+
+     if (codCol==3) // albastru
+    {
+  	// Ambient
+    float ambientStrength = 0.2f;
+    vec3 ambient = ambientStrength * lightColor;
+  	
+    // Diffuse 
+    vec3 normala = normalize(Normal);
+    vec3 lightDir = normalize(inLightPos - FragPos);
+    //vec3 lightDir = normalize(dir); // cazul unei surse directionale
+    float diff = max(dot(normala, lightDir), 0.0);
+    vec3 diffuse = diff * lightColor;
+    
+    // Specular
+    float specularStrength = 0.5f;
+    vec3 viewDir = normalize(inViewPos - FragPos);//vector catre observator normalizat (V)
+    vec3 reflectDir = reflect(-lightDir, normala); // reflexia razei de lumina (R)
+    float spec = pow(max(dot(viewDir, reflectDir), 0.0), 1);
+    vec3 specular = specularStrength * spec * lightColor;  
+    vec3 emission=vec3(0.0, 0.0, 0.0);
+    vec3 result = emission+(ambient + diffuse + specular) * vec3(0.0f, 0.0f, 1.0f);
+	out_Color = vec4(result, 1.0f);
+    }
+
+     if (codCol==4) // galben
```

Comment:
There is too much duplicate code here, use the if statement only to set the color.